### PR TITLE
dont include data in kdtree unless has valid lat/lng pair #7

### PR DESCRIPTION
**Pull request by bloodfire91**

### CODE REVIEW

File: geocoder.js [hunk 1]
```diff
@@ -408,14 +408,27 @@ var geocoder = {
     var data = [];
     var lenI = GEONAMES_COLUMNS.length;
     var that = this;
+
+    //
```
Comment (tomayac) [inline geocoder.js]: Please remove.

File: geocoder.js [hunk 2]
```diff
@@ -408,14 +408,27 @@ var geocoder = {
...
       for (var i = 0; i < lenI; i++) {
         var column = line[i] || null;
         lineObj[GEONAMES_COLUMNS[i]] = column;
       }
-      data.push(lineObj);
+      
+      //
+      var lng = lineObj[GEONAMES_COLUMNS[latitudeIndex]];
+      var lat = lineObj[GEONAMES_COLUMNS[longitudeIndex]];
+      if(lng !== null && lng !== undefined && !isNaN(lng) && lat !== null && lat !== undefined && !isNaN(lat)) {
```
Comment (tomayac) [inline geocoder.js]: Please wrap at 80 characters.

**Comment by tomayac** [comment]
Thanks for the Pull Request :-) Could you please address the two comments, and I am happy to merge. Does this close Issue https://github.com/tomayac/local-reverse-geocoder/issues/7?

**Comment by bloodfire91** [comment]
It works for all the cases I mentioned except Tokyo. If I set the max results to 2 however it works, so not sure whats up in that case.

**Comment by tomayac** [comment]
Strange. Did you investigate the kd tree in these cases?

**Comment by bloodfire91** [comment]
No I havent looked into the kd tree; I will probably have to look into this at some point

File: geocoder.js [hunk 3]
```diff
@@ -422,7 +422,8 @@ var geocoder = {
       var lng = lineObj[GEONAMES_COLUMNS[latitudeIndex]];
       var lat = lineObj[GEONAMES_COLUMNS[longitudeIndex]];
       //dont add lineObj without lat/lng pair
-      if(lng !== null && lng !== undefined && !isNaN(lng) && lat !== null && lat !== undefined && !isNaN(lat)) {
+      if(lng !== null && lng !== undefined && !isNaN(lng) && lat !== null
```
Comment (tomayac) [inline geocoder.js]: Sorry to be a pain on this, could you use the indentation below:
```javascript
if (lng !== null && lng !== undefined && !isNaN(lng) &&
    lat !== null && lat !== undefined && !isNaN(lat)) {
  data.push(lineObj);      
```
Thanks!

Comment (bloodfire91) [inline geocoder.js]: sure thing. is there a style guide you follow btw? I'm kind of new to javascript

Comment (tomayac) [inline geocoder.js]: Thanks for your patience. The code should follow [Google's style guide](https://google.github.io/styleguide/javascriptguide.xml). To automate style linting, you can use [jscs](http://jscs.info/overview) with the [google preset](http://jscs.info/overview#cli).

**Comment by tomayac** [comment]
Thanks a ton for making local-reverse-geocoder better! Merged…

**Comment by bloodfire91** [comment]
sweet! thx man!

**Comment by tomayac** [comment]
No, thank **you**! https://github.com/tomayac/local-reverse-geocoder/commit/7d1052bf93a45b373e5268f8021ec7978d13d7b0

**Pull request merged**